addons/point_of_sale/models/pos_session.py

```diff
@@ -2083,16 +2093,19 @@ def get_pos_ui_product_pricelist_item_by_product(self, product_tmpl_ids, product
         params = self._load_data_params(self.config_id)
         pricelist_params = params['product.pricelist']
         pricelist_item_params = params['product.pricelist.item']
-        pricelists = self.env['product.pricelist'].search(**pricelist_params)
 
         pricelist_item_domain = [
-            ('pricelist_id', 'in', [p.id for p in pricelists]),
             '|',
             '&', ('product_id', '=', False), ('product_tmpl_id', 'in', product_tmpl_ids),
             ('product_id', 'in', product_ids)]
 
-        pricelist_item = self.env['product.pricelist.item'].search_read(pricelist_item_domain, pricelist_item_params['fields'], load=False)
-        return {'product.pricelist.item': pricelist_item}
+        pricelist_item = self.env['product.pricelist.item'].search(pricelist_item_domain)
+        pricelist = pricelist_item.pricelist_id
+
+        return {
+            'product.pricelist.item': pricelist_item.read(pricelist_item_params['fields'], load=False),
+            'product.pricelist': pricelist.read(pricelist_params['fields'], load=False)
+        }
```

Comment:
with this, we'll load unnecessary pricelists. make the method accept a param called 'pricelist_ids'. If provided, use it in the domain for the 'pricelist_item_domain', if not provided, use the pricelists from the config.